src/js/tracks/text-track-cue-list.js

```diff
@@ -57,14 +106,47 @@ TextTrackCueList.prototype.setCues_ = function(cues) {
       });
     }
   };
+  defineProp.call(this, this.cues_.length-1);
+  this.length_ = this.cues_.length;
+};
+
+TextTrackCueList.prototype.removeCue_ = function(removeCue) {
+  let removed = false;
+
+  for (let i = 0, l = this.cues_.length; i < l; i++) {
+    let cue = this.cues_[i];
+    if (cue === removeCue) {
+      this.cues_.splice(i, 1);
+      removed = true;
+    }
+  }
 
-  if (oldLength < l) {
-    i = oldLength;
+  if (removed) {
+    this.length_ = this.cues_.length;
+    if(this.optimized_){
+      this.optimizeAllCues_();
+    }
+  }
+};
+
+TextTrackCueList.prototype.getActiveCuesByTime = function(time) {
+  let cues = this.quickCues_[time/100|0]||[];
+  let active = [];
 
-    for(; i < l; i++) {
-      defineProp.call(this, i);
+  for (let i = 0, l = cues.length; i < l; i++) {
+    let cue = cues[i];
```

Comment:
is it possible that a cue was added to the track not ended up not being optimized so it doesn't show up? Is it even a case we should worry about?